augur/dates/__init__.py

```diff
@@ -121,6 +121,12 @@ def get_numerical_date_from_value(value, fmt=None, min_max_year=None):
         except InvalidDate as error:
             raise AugurError(str(error)) from error
         return [treetime.utils.numeric_date(d) for d in ambig_date]
+    if value.startswith('[') and value.endswith(']') and len(value[1:-1].split(':'))==2:
```

Comment:
For added context, this is the corresponding logic in TreeTime: [treetime/utils.py#L305-L312](https://github.com/neherlab/treetime/blob/52e7cd106a9e0b9a8ea661f213e7e5338e1165cd/treetime/utils.py#L305-L312)